cmd/snap-seccomp/versioninfo.go

```diff
@@ -0,0 +1,503 @@
+	"s390_runtime_instr",
+	"s390_sthyi",
+	"sched_getaffinity",
+	"sched_getattr",
+	"sched_getparam",
+	"sched_get_priority_max",
+	"sched_get_priority_min",
+	"sched_getscheduler",
+	"sched_rr_get_interval",
+	"sched_setaffinity",
+	"sched_setattr",
+	"sched_setparam",
+	"sched_setscheduler",
+	"sched_yield",
+	"seccomp",
+	"security",
+	"select",
+	"semctl",
+	"semget",
+	"semop",
+	"semtimedop",
+	"send",
+	"sendfile",
+	"sendfile64",
+	"sendmmsg",
+	"sendmsg",
+	"sendto",
+	"setdomainname",
+	"setfsgid",
+	"setfsgid32",
+	"setfsuid",
+	"setfsuid32",
+	"setgid",
+	"setgid32",
+	"setgroups",
+	"setgroups32",
+	"sethostname",
+	"setitimer",
+	"set_mempolicy",
+	"setns",
+	"setpgid",
+	"setpriority",
+	"setregid",
+	"setregid32",
+	"setresgid",
+	"setresgid32",
+	"setresuid",
+	"setresuid32",
+	"setreuid",
+	"setreuid32",
+	"setrlimit",
+	"set_robust_list",
+	"setsid",
+	"setsockopt",
+	"set_thread_area",
+	"set_tid_address",
+	"settimeofday",
+	"set_tls",
+	"setuid",
+	"setuid32",
+	"setxattr",
+	"sgetmask",
+	"shmat",
+	"shmctl",
+	"shmdt",
+	"shmget",
+	"shutdown",
+	"sigaction",
+	"sigaltstack",
+	"signal",
+	"signalfd",
+	"signalfd4",
+	"sigpending",
+	"sigprocmask",
+	"sigreturn",
+	"sigsuspend",
+	"socket",
+	"socketcall",
+	"socketpair",
+	"splice",
+	"spu_create",
+	"spu_run",
+	"ssetmask",
+	"stat",
+	"stat64",
+	"statfs",
+	"statfs64",
+	"statx",
+	"stime",
+	"stty",
+	"subpage_prot",
+	"swapcontext",
+	"swapoff",
+	"swapon",
+	"switch_endian",
+	"symlink",
+	"symlinkat",
+	"sync",
+	"sync_file_range",
+	"sync_file_range2",
+	"syncfs",
+	"syscall",
+	"_sysctl",
+	"sys_debug_setcontext",
+	"sysfs",
+	"sysinfo",
+	"syslog",
+	"sysmips",
+	"tee",
+	"tgkill",
+	"time",
+	"timer_create",
+	"timer_delete",
+	"timerfd",
+	"timerfd_create",
+	"timerfd_gettime",
+	"timerfd_settime",
+	"timer_getoverrun",
+	"timer_gettime",
+	"timer_settime",
+	"times",
+	"tkill",
+	"truncate",
+	"truncate64",
+	"tuxcall",
+	"ugetrlimit",
+	"ulimit",
+	"umask",
+	"umount",
+	"umount2",
+	"uname",
+	"unlink",
+	"unlinkat",
+	"unshare",
+	"uselib",
+	"userfaultfd",
+	"usr26",
+	"usr32",
+	"ustat",
+	"utime",
+	"utimensat",
+	"utimes",
+	"vfork",
+	"vhangup",
+	"vm86",
+	"vm86old",
+	"vmsplice",
+	"vserver",
+	"wait4",
+	"waitid",
+	"waitpid",
+	"write",
+	"writev",
+}
+
+func versionInfo() string {
+	// Calculate the checksum of all syscall names supported by libseccomp
+	// library. We add that to the version info to cover the case when
+	// libseccomp version does not change, but the set of supported syscalls
+	// does due to distro patches.
+	sh := sha256.New()
+	newline := []byte("\n")
+	for _, syscallName := range seccompSyscalls {
+		if _, err := seccomp.GetSyscallFromName(syscallName); err != nil {
+			// syscall is unsupported by this version of libseccomp
+			continue
+		}
+		sh.Write([]byte(syscallName))
+		sh.Write(newline)
+	}
+
+	major, minor, micro := seccomp.GetLibraryVersion()
+
+	return fmt.Sprintf("%d.%d.%d %x", major, minor, micro, sh.Sum(nil))
```

Comment:
this needs to include also a version of the compiler itself, no?

Reply:
As discussed on IRC, we'll include the build-id of snap-seccomp binary.